### DISCUSSION TITLE
Do _comp and _component mean same thing in INSMomentumBase.C?

### DISCUSSION THREAD

**Discussion by Minjiang-Zhu** [idaholab/moose#26715] [Q&A General] [answered]
As title says.

**Comment by lindsayad** [comment, marked as answer]
No they don't. `_component` indicates the momentum component we are building the residual/Jacobian rows for. `comp` is used when specifying which momentum component we are building the Jacobian columns for, so how does the momentum component `_component` depend on the momentum component `comp`. So you can think of `_component` denoting the row, and `comp` the column in a coupling matrix
